### PR TITLE
Supporting `posonlyargs` in Python

### LINKED ISSUE #1637: Support PEP 570 – Python Positional-Only Parameters
See https://peps.python.org/pep-0570/. It seems they end up in `posonlyargs` instead of `args` and this currently crashes if this is used in a method declaration.

### PR DESCRIPTION
**Pull request by oxisto**
Fixes #1637

### CODE REVIEW

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandlerTest.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        var func = result.functions["pos_only_and_args"]
+        assertNotNull(func)
+
+        val list = listOf("a", "b", "c")
+        list.forEachIndexed { idx, name ->
+            var param = func.parameterEdges.firstOrNull { it.end.name.localName == name }
+            assertNotNull(param, "$name should not be empty")
+            assertEquals(idx, param.getProperty(Properties.INDEX))
+        }
```
Comment (maximiliankaul) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandlerTest.kt:64]: Maybe add a check for the posonly modifier?

**Review by maximiliankaul — APPROVED** [review]

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt [hunk 2]
```diff
@@ -237,8 +237,6 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
 
         frontend.scopeManager.enterScope(cls)
 
-        stmt.keywords.map { TODO() }
```
[the remark this answers is not in the document]
Comment (oxisto) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt:240]: Not really sure where :(

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandlerTest.kt (hunk 1 above)
Comment (oxisto) [reply cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandlerTest.kt:71]: done

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (oxisto) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt:240]: I added it to the current translation unit

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguage.kt [hunk 3]
```diff
@@ -159,4 +159,14 @@ class PythonLanguage :
...
     }
+
+    companion object {
+        /**
+         * This is a "modifier" to differentiate parameters in functions that are "positional" only.
+         * This information will be stored in
+         * [de.fraunhofer.aisec.cpg.graph.declarations.ParameterDeclaration.modifiers] so that we
+         * can use is later in call resolving.
+         */
+        const val MODIFIER_POSITIONAL_ONLY_ARGUMENT = "posonlyarg"
```
[the remark this answers is not in the document]
Comment (oxisto) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguage.kt:170]: I added MODIFIER_KEYWORD_ONLY_ARGUMENT.

**Pull request merged**